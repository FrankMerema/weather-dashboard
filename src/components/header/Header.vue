<template>
  <header class="ws-header">
    <ws-icon icon="logo" :chameleon="true" />
    <h2>Merema's Weather</h2>
  </header>
</template>

<script>
import WsIcon from '~/components/icons/Icon';

export default {
  components: {
    WsIcon
  }
};
</script>

<style lang="scss" scoped>
.ws-header {
  align-items: center;
  background: $ws-background-to-color;
  box-shadow: $ws-black 0 1px 24px 0;
  color: $ws-white;
  display: flex;
  justify-content: center;
  padding: $ws-gutter-size-small $ws-gutter-size;
  z-index: 1;

  h2 {
    flex: 1;
    margin: 0;
  }

  .ws-icon {
    @include size($ws-gutter-size-small * 3);
    margin: $ws-gutter-size-small $ws-gutter-size-small $ws-gutter-size-small 0;
  }
}
</style>
