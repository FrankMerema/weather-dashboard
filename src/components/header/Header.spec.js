import { mount } from '@vue/test-utils';
import Header from './Header';

describe('Header', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Header);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
});

describe('Header snapshot', () => {
  test('render Header correctly', () => {
    const wrapper = mount(Header);
    expect(wrapper.element).toMatchSnapshot();
  });
});
