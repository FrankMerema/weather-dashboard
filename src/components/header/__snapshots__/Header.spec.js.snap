// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot render Header correctly 1`] = `
<header
  class="ws-header"
>
  <svg
    class="ws-icon chameleon"
  >
    <use
      xlink:href="#ws-logo"
    />
  </svg>
   
  <h2>
    Merema's Weather
  </h2>
</header>
`;
